### PR TITLE
Stats icon: Fix a fatal when the first column is "comments"

### PR DESCRIPTION
**Pull request by BogdanUngureanu**
Fixes DOTCOM-14263

In certain conditions, when the post list contains "comments" as the first column, the stats icon column was throwing a PHP Fatal error.

This PR fixes this issue by handling this scenario. I couldn't reproduce it locally on a test site. Instead, I've wrote a unit test that validates the fix.

## Proposed changes:

* Bug fix

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Test passes

### CODE REVIEW

File: projects/packages/stats-admin/tests/php/Admin_Post_List_Test.php [hunk 1]
```diff
@@ -51,6 +51,36 @@ public function test_add_stats_post_table_adds_stats_column() {
 		$this->assertEquals( 'Stats', $columns_with_stats['stats'] );
 	}
 
+	/**
+	 * Test if the stats column is added when comments is the first column of the table.
+	 *
+	 * @return void
+	 */
+	public function test_adds_stats_column_when_comments_is_the_first_column() {
```
Comment (BogdanUngureanu) [inline projects/packages/stats-admin/tests/php/Admin_Post_List_Test.php:59]: I could write a data provider to have a single test with two datasets and avoid having two identical tests. But... I'm not sure if it's worth the effort.

**Review by tbradsha — APPROVED** [review]
Looks like this should address the edge case in question. Thanks!
